lib/http/HttpRequestEncoder.cpp

```diff
@@ -45,6 +45,11 @@ namespace MAT_NS_BEGIN {
         ctx->httpRequest->SetUrl(m_config.GetCollectorUrl());
 
         ctx->httpRequest->GetHeaders().set("Expect", "100-continue");
+
+#ifdef HAVE_MAT_EVT_TRACKINGID
+        ctx->httpRequest->GetHeaders().set("Trace-Id", ctx->traceId);
```

Comment:
As per the w3c trace context specs, the trace_id should be part of HTTP header "traceparent" - 
https://www.w3.org/TR/trace-context/#traceparent-header

Header["traceparent"] =  `version`-`trace-id`-`parent-id`-`trace-flags`

I don't have strong opinion for now, as we are not sending it across the wire.

Reply:
Currently we need traceId only, hence logging that only. However, let me know if this change is must.